### ISSUE TITLE
thread main panicked after piping a nu script into where

### ISSUE THREAD

**Issue by Yethal** [nushell/nushell#3893]
### Describe the bug

After running a nu script that outputs a table and piping it into 'where' clause following stacktrace is printed:

```error: Type Error
  ┌─ shell:6:18
  │
6 │ RUST_BACKTRACE=1 nuschiit|where availability == 'In Stock'
  │                  ^^^^^^^^ Expected row or table, found string

/home/yethal> thread 'main' panicked at 'failed printing to stdout: Broken pipe (os error 32)', library/std/src/io/stdio.rs:935:9
stack backtrace:
   0: rust_begin_unwind
             at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/std/src/panicking.rs:493:5
   1: std::panicking::begin_panic_fmt
             at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/std/src/panicking.rs:435:5
   2: std::io::stdio::print_to
             at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/std/src/io/stdio.rs:935:9
   3: std::io::stdio::_print
             at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/std/src/io/stdio.rs:947:5
   4: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
   5: std::thread::local::LocalKey<T>::with
   6: futures_executor::local_pool::block_on
   7: <nu_command::commands::viewers::table::command::Command as nu_engine::whole_stream_command::WholeStreamCommand>::run
   8: nu_engine::whole_stream_command::Command::run
   9: nu_command::commands::viewers::autoview::command::autoview
  10: <nu_command::commands::viewers::autoview::command::Command as nu_engine::whole_stream_command::WholeStreamCommand>::run
  11: nu_engine::whole_stream_command::Command::run
  12: nu_engine::script::process_script
  13: nu_engine::script::run_script_standalone
  14: nu_cli::cli::run_script_file
  15: nu_cli::app::App::main
  16: nu_cli::app::App::run
  17: nu::main
```

### How to reproduce

1. Save script from PR https://github.com/nushell/nu_scripts/pull/83 as script
1. Run ./script|where availability =='In Stock'
1. Observe the nushell panicking

### Expected behavior

I expected nu to filter the table and output only matching rows

### Configuration

> version|pivot
───┬────────────────────┬───────────────────────────────────────────────────────────────────────
 # │ Column0            │ Column1
───┼────────────────────┼───────────────────────────────────────────────────────────────────────
 0 │ version            │ 0.35.0
 1 │ build_os           │ linux-x86_64
 2 │ rust_version       │ rustc 1.53.0 (53cb7b09b 2021-06-17)
 3 │ rust_channel       │ stable-x86_64-unknown-linux-gnu (default)
 4 │ cargo_version      │ cargo 1.53.0 (4369396ce 2021-04-27)
 5 │ pkg_version        │ 0.35.0
 6 │ build_time         │ 2021-08-04 17:26:23
 7 │ build_rust_channel │ release
 8 │ features           │ clipboard-cli, ctrlc, dataframe, default, rustyline, term, trash,
   │                    │ uuid, which, zip
 9 │ installed_plugins  │ binaryview, chart bar, chart line, fetch, from bson, from sqlite,
   │                    │ inc, match, post, ps, query json, s3, selector, start, sys, textview,
   │                    │ to bson, to sqlite, tree, xpath
───┴────────────────────┴───────────────────────────────────────────────────────────────────────

**Comment by jntrnr** [comment]
When you run a script, if you want to get the the structured output you'll need to use something like `source ./script.nu | where ... ` I believe. Otherwise, you'll be getting the strings as it will run the script from outside of Nushell.

That said, we shouldn't be panicking here.

**Comment by andrasio** [comment]
@Yethal Can we look at the code in the `nuschiit` file?

**Comment by andrasio** [comment]
@Yethal 

Noticed your first step. For your second step though, you are doing:

are you running `./script|where availability =='In Stock'` inside a nu session?

**Comment by Yethal** [comment]
@andrasio Yes, the nuschiit script (named liked that because I already had schiit script that did the same thing using xmllint) is the same as the example I added
I am running it inside a nu session.

**Comment by Yethal** [comment]
I think this might be instance of https://github.com/nushell/nushell/issues/3551

**Comment by jntrnr** [comment]
The panicking part of this seems to be fixed in the latest main.

We should use a separate bug to track any automatic structure detection between external to internal. It's an area we've explored some, but we don't want to do things here that would surprise folks.

Assuming `nu` is like `source` is another thing we could explore. Again, that should also be a separate thread as there are some complications there we should work through.

Closing as I think the original issue is fixed now.
